spec/lib/freyja/persister_spec.rb

```diff
@@ -0,0 +1,483 @@
+            expect(updated_resource[Valkyrie::Persistence::Attributes::OPTIMISTIC_LOCK]).not_to be_empty
+          end
+        end
+      end
+
+      describe "#save_all" do
+        context "when creating multiple resources" do
+          it "returns an array of resources with their system-generated optimistic locking attributes" do
+            resource1 = MyLockingResource.new(title: ["My Locked Resource 1"])
+            resource2 = MyLockingResource.new(title: ["My Locked Resource 2"])
+            resource3 = MyLockingResource.new(title: ["My Locked Resource 3"])
+            saved_resources = persister.save_all(resources: [resource1, resource2, resource3])
+            saved_resources.each do |saved_resource|
+              expect(saved_resource[Valkyrie::Persistence::Attributes::OPTIMISTIC_LOCK]).not_to be_empty
+            end
+          end
+        end
+
+        context "when updating multiple resources that all have a correct lock token" do
+          it "saves the resources and returns them with updated values of the optimistic locking attribute" do
+            resource1 = MyLockingResource.new(title: ["My Locked Resource 1"])
+            resource2 = MyLockingResource.new(title: ["My Locked Resource 2"])
+            resource3 = MyLockingResource.new(title: ["My Locked Resource 3"])
+            saved_resources = persister.save_all(resources: [resource1, resource2, resource3])
+            initial_lock_tokens = saved_resources.map do |r|
+              r[Valkyrie::Persistence::Attributes::OPTIMISTIC_LOCK]
+            end
+            updated_resources = persister.save_all(resources: saved_resources)
+            updated_lock_tokens = updated_resources.map do |r|
+              r[Valkyrie::Persistence::Attributes::OPTIMISTIC_LOCK]
+            end
+            expect(initial_lock_tokens & updated_lock_tokens).to be_empty
+          end
+        end
+
+        context "when one of the resources has an incorrect lock token" do
+          it "raises a Valkyrie::Persistence::StaleObjectError" do
+            resource1 = MyLockingResource.new(title: ["My Locked Resource 1"])
+            resource2 = MyLockingResource.new(title: ["My Locked Resource 2"])
+            resource3 = MyLockingResource.new(title: ["My Locked Resource 3"])
+            resource1, resource2, resource3 = persister.save_all(resources: [resource1, resource2, resource3])
+            # update a resource in the datastore to make its token stale
+            persister.save(resource: resource2)
+
+            expect { persister.save_all(resources: [resource1, resource2, resource3]) }
+              .to raise_error(Valkyrie::Persistence::StaleObjectError)
+          end
+        end
+      end
+    end
+
+    context 'ordered properties' do
+      xit "orders string values and returns them in the appropriate order" do
+        validate_order ["a", "b", "a"]
+      end
```

Comment:
All of the specs in the ordered properties section are skipped. What's the reason for that?

Reply:
no sure why these were skipped, they work fine. I brought them all back